scripts/tank_cmd.py

```diff
@@ -1230,7 +1254,148 @@ def run_engine_cmd(log, pipeline_config_root, context_items, command, using_cwd,
         return tank_command.run_action(log, tk, ctx, command, args)
 
 
+def _extract_args(cmd_line, *args):
+    """
+    Extract specified arguments and sorts them. Also removes them from the
+    command line.
+
+    :param cmd_line: Array of command line arguments.
+    :param args: Argument list to extract
+
+    :returns: A list of key, value tuples representing each arguments.
+    """
+    # Find all instances of each argument in the command line
+    found_args = []
+    for cmd_line_token in cmd_line:
+        for arg in args:
+            if cmd_line_token.startswith("--%s=" % arg):
+                found_args.append(cmd_line_token)
+
+    # Strip all these arguments from the command line.
+    for a in found_args:
+        cmd_line.remove(a)
```

Comment:
Rather than editing in place like this, I think we should return a new value:

```
(params, new_args) = _extract_args(args, ["foo", "bar", "baz"]
```

In my opinion, keeping input arguments immutable leads to less confusion. Unless you have good reasons to edit an input parameter of course. Const correctness in C++ springs to mind :)